frontend/src/components/TerminalSplitpanes.vue

```diff
@@ -0,0 +1,288 @@
+    v-shortkey.once="{bottom: ['ctrl', 'shift', 'h'], right: ['ctrl', 'shift', 'v']}"
+    @dropped="droppedAt"
+    @shortkey="addFromShortkey"
+  >
+    <g-splitpane
+      v-if="splitpaneTree"
+      :splitpaneTree="splitpaneTree"
+      ref="splitpane"
+    >
+      <template v-slot="{item}">
+        <slot v-if="item.data.type === 'SLOT_ITEM'" v-bind:item="item"></slot>
+        <g-terminal
+          v-else
+          :uuid="item.uuid"
+          :data="item.data"
+          @terminated="onTermination(item)"
+          @split="orientation => onSplit(item, orientation)"
+        ></g-terminal>
+      </template>
+    </g-splitpane>
+    <create-terminal-session-dialog
+      ref="newTerminal"
+      :name="name"
+      :namespace="namespace"
+    ></create-terminal-session-dialog>
+  </div>
+</template>
+
+<script>
+
+import { mapGetters } from 'vuex'
+import every from 'lodash/every'
+import get from 'lodash/get'
+import filter from 'lodash/filter'
+import includes from 'lodash/includes'
+import map from 'lodash/map'
+import merge from 'lodash/merge'
+import cloneDeep from 'lodash/cloneDeep'
+import difference from 'lodash/difference'
+import GSplitpane from '@/components/GSplitpane'
+import GTerminal from '@/components/GTerminal'
+import CreateTerminalSessionDialog from '@/components/dialogs/CreateTerminalSessionDialog'
+import { TargetEnum, routeName } from '@/utils'
+import { listTerminalSessions } from '@/utils/api'
+import { GSymbolTree, Leaf, PositionEnum } from '@/lib/g-symbol-tree'
+
+import 'splitpanes/dist/splitpanes.css'
+
+function terminatedSessionIds (uuids, terminals) {
+  const terminalSessionIds = map(terminals, 'metadata.identifier')
+  return difference(uuids, terminalSessionIds)
+}
+
+export default {
+  components: {
+    GSplitpane,
+    GTerminal,
+    CreateTerminalSessionDialog
+  },
+  props: {
+    name: {
+      type: String
+    },
+    namespace: {
+      type: String
+    },
+    target: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      tree: new GSymbolTree(),
+      splitpaneTree: undefined // splitpaneTree is a json object representation of the GSymbolTree`this.tree`
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'focusedElementId',
+      'hasControlPlaneTerminalAccess',
+      'hasShootTerminalAccess'
+    ]),
+    terminalCoordinates () {
+      const coordinates = {
+        name: this.name,
+        namespace: this.namespace,
+        target: this.target
+      }
+      return coordinates
+    },
+    storeKey () {
+      const { name, namespace } = this.terminalCoordinates
+      const route = routeName(this.$route)
+      return `${route}--${namespace}--${name}`
+    },
+    slotItemUUIds () {
+      const slotItems = filter(this.tree.items(), ['data.type', 'SLOT_ITEM'])
+      return map(slotItems, 'uuid')
+    }
+  },
+  methods: {
+    updateSplitpaneTree () {
+      this.splitpaneTree = this.tree.toJSON(this.tree.root)
+
+      this.store()
```

Comment:
maybe rename this method. This method stores the current state into the local storage, right? Its a bit confusing as store usually refers to the Vuex store. maybe rename to updateLocalStorage or maybe you find a better name

Reply:
renamed to `storeSplitpaneTree`